scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+        try {
+            PreparedStatement ps = con.prepareStatement(deleteBookByIdQuery);
+            ps.setString(1, bookId);
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                response = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            response += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return response;
+    }
+
+    @Override
+    public String addBook(Book book) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(addBookQuery);
+            ps.setString(1, book.getBarcode());
+            ps.setString(2, book.getName());
+            ps.setString(3, book.getAuthor());
+            ps.setDouble(4, book.getPrice());
+            ps.setInt(5, book.getQuantity());
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                responseCode = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public String updateBookQtyById(String bookId, int quantity) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(updateBookQtyByIdQuery);
+            ps.setInt(1, quantity);
+            ps.setString(2, bookId);
+            ps.executeUpdate();
+            responseCode = ResponseCode.SUCCESS.name();
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public List<Book> getBooksByCommaSeperatedBookIds(String commaSeperatedBookIds) throws StoreException {
+        List<Book> books = new ArrayList<Book>();
+        Connection con = DBUtil.getConnection();
+        try {
+            String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+                    + " WHERE " +
+                    BooksDBConstants.COLUMN_BARCODE + " IN ( " + commaSeperatedBookIds + " )";
+            PreparedStatement ps = con.prepareStatement(getBooksByCommaSeperatedBookIdsQuery);
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                String bCode = rs.getString(1);
+                String bName = rs.getString(2);
+                String bAuthor = rs.getString(3);
+                int bPrice = rs.getInt(4);
+                int bQty = rs.getInt(5);
+
+                Book book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+
+        }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider handling SQLException appropriately</b></div>
<div id="fix">

The `SQLException` catch block is empty which silently swallows exceptions. Consider logging the exception or propagating it to the caller. A similar issue was also found in scr/main/java/bittercode/service/impl/BookServiceImpl.java (line 83-85).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -58,3 +58,3 @@
         } catch (SQLException e) {
 -
 +            throw new StoreException("Failed to get book by ID: " + e.getMessage());
         }
```


```suggestion
       } catch (SQLException e) {
            throw new StoreException("Failed to get book by ID: " + e.getMessage());
       }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/7#issuecomment-2527740799>#4862f0</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged